### ISSUE TITLE
Zeppelin installation error

### ISSUE THREAD

**Issue by notsodirewolf** [OpenZeppelin/zeppelin-solidity#209]
Hi all,
I am getting the following error:
```
>truffle install zeppelin
Error: Could not find object at hash 'QmYt6ZfD1VNmKp85TeteEcsgaxHzP2gTSKupHxHMVgcHYb' in 5000ms
    at Timeout._onTimeout (/usr/local/lib/node_modules/truffle/node_modules/ethpm/lib/hosts/ipfshost.js:47:14)
    at ontimeout (timers.js:386:14)
    at tryOnTimeout (timers.js:250:5)
    at Timer.listOnTimeout (timers.js:214:5)
```
Has anyone had experience with this?

**Comment by frangio** [comment]
It's an issue with IPFS, the storage used for the Ethereum Package Registry.

I recommend installing via npm for now: `npm install zeppelin-solidity`.

**Comment by notsodirewolf** [comment]
Hmm, after I execute
npm install zeppelin-solidity

I get the following error:
dhcp-v229-197:eqc-token huax$ npm install zeppelin-solidity
npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/Cellar/node/7.10.0/bin/node" "/usr/local/bin/npm"
"install" "zeppelin-solidity"
npm ERR! node v7.10.0
npm ERR! npm  v4.2.0
npm ERR! code ENOSELF

npm ERR! Refusing to install package with name "zeppelin-solidity" under a
package
npm ERR! also called "zeppelin-solidity". Did you name your project the same
npm ERR! as the dependency you're installing?

**Comment by maraoz** [comment]
Just reported this on `truffle`, FYI: https://github.com/trufflesuite/truffle/issues/378

**Comment by maraoz** [comment]
Another occurrence here: https://github.com/OpenZeppelin/zeppelin-solidity/issues/96#issuecomment-299595486

**Comment by amexperts** [comment]
Having same issue on Mac OSX

**Comment by maraoz** [comment]
@notsodirewolf @amexperts @ProphetDaniel @frangio just published a new release of the framework. Can you attempt to install it again and report if you have any problems? 
Just run `truffle install zeppelin` in your project. Thanks!

**Comment by amexperts** [comment]
@maraoz - same problem for me I am afraid.

**Comment by dukei** [comment]
It does not help, still the same error. 

Error: Could not find object at hash 'QmPhHVUuwMkkGMg7TWQw7eFANvTrWugpeXqQ7SgkawPN2b' in 5000ms

Still https://ipfs.io/ipfs/QmPhHVUuwMkkGMg7TWQw7eFANvTrWugpeXqQ7SgkawPN2b is available but could not be retrieved by truffle

**Comment by notsodirewolf** [comment]
Me too, still same error.

**Comment by maraoz** [comment]
Damn. We'll have to revert to `npm` for the time being then :(

**Comment by maraoz** [comment]
@notsodirewolf @amexperts @ProphetDaniel @frangio Thanks a bunch for your patience. We've reverted to `npm` for now. Can you check the new installation instructions below work for you?

https://github.com/OpenZeppelin/zeppelin-solidity#getting-started

@tcoulter I'm sad to do this but I think you understand. We loved being early adopters of ethpm and we'll be eager to jump back to using it once it's reliable. :)

**Comment by tcoulter** [comment]
Absolutely understand. Once the issues are figured out I'd recommend
deploying on both EthPM and NPM to reach the widest audience. Additionally,
I envision the internals of EthPM will change once SWARM is complete and
file persistence is economically guaranteed.

I'm continuing to look at this issue and will let you know what I find.

**Comment by maraoz** [comment]
Any luck guys?

**Comment by tcoulter** [comment]
Yep. I've isolated the issue, but to fix it will require some coding (which I'm working on now).

Background: I was wrong. The issue we're having with Zeppelin *wasn't* the issue where data was lost. The results of the issue looked the same at first glance, but instead it was a combination of smaller issues. 

* The first issue is that ethpm-js has a download timeout of 5 seconds per file, and if the file takes longer than five seconds to download, it bails, resulting in the error you guys are seeing.
* The second issue is that, for some reason (I'm still trying to figure it out) the `ipfs-api-js` module ethpm-js uses to download IPFS files is _suuuuuper_ slow, taking more than five seconds to resolve a file. That same file, when downloaded over http, is nearly instant. 

Now some technical details: Because IPFS URIs can denote files **OR** directories of files (:grumble grumble:), in order to be up to spec I need to now walk the IPFS directory and download all the individual files over http. `ipfs-api-js` was doing this for me, but I'm ditching it in favor of downloading files the speedy way. 

So **TL;DR:** Problem isolated, but still working on code to get around it. Oh and no Zeppelin data was lost.

**Comment by amexperts** [comment]
That seems to work. But just a note it is installing into contracts folder now. :)

**Comment by tcoulter** [comment]
So you guys know: The issue has been fixed on Truffle v3.2.3. You can get the latest version by uninstalling and then reinstalling Truffle. Cheers! Thanks for the reports, and appreciate your patience.

**Comment by notsodirewolf** [comment]
Thank you!
